### PR TITLE
refactor delete/find query, fix #148.

### LINKED ISSUE #148: Investigate query method delete operation
For now, our delete of DocumentQuery will find the domains at first, and then delete one by one from the find result document.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Separate deleteAll domains and deleteCollection
* Reuse the deleteAll and findAll implementation.

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java [hunk 1]
```diff
@@ -181,69 +184,40 @@ private DocumentClient getDocumentClient() {
...
-                    + " in database: " + this.databaseName + ", but found " + collections.size());
-        }
+        Assert.notNull(domainClass, "entityClass should not be null");
 
-        final FeedOptions feedOptions = new FeedOptions();
-        feedOptions.setEnableCrossPartitionQuery(true);
-
-        final SqlQuerySpec sqlQuerySpec = new SqlQuerySpec("SELECT * FROM root c");
+        final DocumentQuery query = new DocumentQuery(Criteria.getInstance(CriteriaType.ALL), Sort.unsorted());
+        final List<Document> results = findAllDocuments(query, domainClass, collectionName);
```
Comment (sophiaso) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: test

Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: test++

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java [hunk 2]
```diff
@@ -465,40 +437,54 @@ private boolean isCollectionSupportSortByString(@NonNull DocumentCollection coll
         return indices.stream().anyMatch(isIndexingSupportSortByString());
     }
 
+    private List<Document> findAllDocuments(@NonNull DocumentQuery query, @NonNull Class<?> domainClass,
```
Comment (sophiaso) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: Why the method name is named as `findAllDocuments`? Seems it's not limited with `findAll`?

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java [hunk 3]
```diff
@@ -465,40 +437,54 @@ private boolean isCollectionSupportSortByString(@NonNull DocumentCollection coll
...
+        try {
+            final RequestOptions options = new RequestOptions();
+
+            if (StringUtils.hasText(partitionKeyName)) {
+                options.setPartitionKey(new PartitionKey(document.get(partitionKeyName)));
+            }
+
+            getDocumentClient().deleteDocument(document.getSelfLink(), options);
+        } catch (DocumentClientException e) {
+            throw new DocumentDBAccessException("Failed to delete document %s" + document.getSelfLink(), e);
```
Comment (sophiaso) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: Seems the `%s` will not be used here?

**Pull request merged**